### DISCUSSION TITLE
Is there a way to disable positional arguments altogether ?

### DISCUSSION THREAD

**Discussion by aphecetche** [CLIUtils/CLI11#1095] [Q&A] [answered]
By this I mean something like : 
- make the dash (or dashes) mandatory when declaring an option
- and throw an exception if the parser encounters something that it cannot associate with any option

**Comment by phlptp** [comment]
`->allow_extras(false)`   will generate an exception for unrecognized options including positional ones.   (if you don't define any positionals).

**Reply by aphecetche** [reply]
Thanks. That would solve the second point above only, right ?
For the first one, could this be achieved with subclassing App somehow ?

**Reply by phlptp** [reply, marked as answer]
The first could be achieved by wrapping the `add_option` call and doing the check there or probably slightly less complicated would be using the `get_options(filter)` method with a filter that selects options with the `get_positional()` check.  If that list is not empty throw an error.   That could be done in a method that operates on an app and does the check.  

Curious what the use case for not allowing positionals in that manner is?

**Reply by aphecetche** [reply]
Something like : 

```
void myparse(CLI::App& app, int argc, char** argv)
{
  app.parse(argc, argv);
  // disallow positional arguments
  auto with_pos_arg = app.get_options([](const CLI::Option* opt)
                                      { return opt->get_positional(); });
  if (!with_pos_arg.empty()) {
    throw std::invalid_argument("no positional argument allowed");
  }
}
```

indeed works fine. Thanks.

As for the reason : I'm trying to see if the current parser we are using could be replaced with CLI11. And our current parser simply does not allow positional arguments.
